nodejs/week3/Mealsharing/src/backend/api/meals.js

```diff
@@ -0,0 +1,202 @@
+const knex = require("../database");
+const express = require("express");
+const router = express.Router();
+//get method
+router.get("/meals", async (req, res) => {
+          try {
+                    const allMeals = await knex.from("meal").select("*");
+                    if (allMeals.length == 0) {
+                              res.status(404).send(
+                                        "Sorry data does not exists any more"
+                              );
+                    }
+                    res.send(allMeals);
+          } catch (err) {
+                    throw err;
+          }
+});
+//get method using id
+router.get("/meals/:id", async (req, res) => {
+          try {
+                    const mealId = req.params.id;
+                    const mealWithId = await knex
+                              .from("meal")
+                              .select("*")
+                              .where({ meal_id: mealId });
+                    if (mealWithId.length == 0) {
+                              res.status(404).send(
+                                        "Sorry data does not exists"
+                              );
+                    }
+                    res.send(mealWithId);
+          } catch (err) {
+                    throw err;
+          }
+});
+//post method
+router.post("/meals", async (req, res) => {
+          const query = {
+                    title: req.query.title,
+                    description: req.query.description,
+                    location: req.query.location,
+                    price: req.query.price,
```

Comment:
what happens if the request provides wrong data? e.g. the price is not a number?

A good advice is to never trust data provided by a user (or a client/consumer) of an API or application. You need to validate any type of data that you receive from the outside world so that it doesn't break your application in unexpected ways. If an error occurs you want to communicate to the client of your application what the error is and how they can recover from that error. That helps you troubleshoot issues and guides your development.

(e.g. if I get a 404 not found I know that the id that I'm sending is probably wrong, if I get a 400 bad request I know that there's some parameter in my request that is wrong, etc)